ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Util.hs

```diff
@@ -374,3 +388,48 @@ fib n = round $ phi ** fromIntegral n / sq5
 eitherToMaybe :: Either a b -> Maybe b
 eitherToMaybe (Left _)  = Nothing
 eitherToMaybe (Right x) = Just x
+
+{-------------------------------------------------------------------------------
+  Fuse code, i.e. regions of code that will throw an exception if accessed
+  concurrently.
+-------------------------------------------------------------------------------}
+
+-- | A WithIntactFuse expression is one of which a single instance can be
+-- evaluated at any given moment in the execution of the program.
```

Comment:
```suggestion
-- executed at any given moment in the lifetime of the program.
```
"Evaluation" has an association with pure code for me, but the fuse stuff is specifically about side effects in `IO`.